std/uni.d

```diff
@@ -8131,6 +8185,7 @@ private auto toCaser(alias indexFn, uint maxIdx, alias tableFn, alias asciiConve
         {
             @property auto save()
             {
+                import std.range.primitives : save;
```

Comment:
>I'd suggest to just add popFront, front and friends to that top-level selective import. It's used everywhere anyway

Actually, it isn't.  `popFront` and `front` over arrays from `std.range.primitives` are imported and used in six scopes each.  I might agree with you if it were a lot more than that and definitely if it were ever done at module scope.

>I'm opposed to adding tons of identical fine-grained imports for commonly used symbols especially since the module is already imported by the top-level scope.

As I said, the module is only fully imported at the top level temporarily, because of a long-standing compiler bug that has been fixed but we're not enforcing yet, to allow users time [to update their code using the current import warnings](http://www.schveiguy.com/blog/2016/03/import-changes-in-d-2-071/).  The goal is to only have those five symbols that are currently commented out be imported at module scope, but that cannot be turned on _yet_.  Once it is, these symbols won't be imported at module scope anymore.

I think one big advantage of fine-grained imports is explicitly documenting where those range functions are actually coming from.  If I'm quickly reading code and see that import, I know it's the range function for arrays being called.  Otherwise, I have to figure out the type of that particular range and go track down its specialized range implementation if I want to see what it's actually doing.  That kind of documentation saves time, particularly for templated code that may call different types of ranges.